### PR TITLE
Fix broken doc signatures and links

### PR DESCRIPTION
**Pull request by garazdawi**
This PR removes broken doc signatures that were generated as part of the erl_docgen to ex_doc migration.

It also removes all anchors at the end of a documentation attribute as those links to those anchors almost always pointed to an incorrect place in the docs.

### CODE REVIEW

File: lib/eldap/src/eldap.erl [hunk 1]
```diff
@@ -1986,7 +1980,7 @@ paged_result_control(PageSize, Cookie) when is_integer(PageSize) ->
 %%% --------------------------------------------------------------------
 
 -doc """
-paged_result_cookie(SearchResult) -> binary()
+paged_result_cookie(SearchResult)
```
Comment (dgud) [inline lib/eldap/src/eldap.erl:1983]: Remove?

File: lib/et/src/et_selector.erl [hunk 2]
```diff
@@ -52,7 +52,7 @@
 %%   integer() - explicit detail level of tracing
 %%----------------------------------------------------------------------
 -doc """
-make_pattern(RawPattern) -> TracePattern
+make_pattern(RawPattern)
```
Comment (dgud) [inline lib/et/src/et_selector.erl:55]: Remove

File: lib/et/src/et_selector.erl [hunk 3]
```diff
@@ -115,7 +115,7 @@ make_pattern({Mod, Pattern}) when is_atom(Mod) ->
 %% accordingly with erlang:trace_pattern/2.
 %%----------------------------------------------------------------------
 -doc """
-change_pattern(Pattern) -> ok
+change_pattern(Pattern)
```
Comment (dgud) [inline lib/et/src/et_selector.erl:118]: Remove

File: lib/observer/src/ttb.erl [hunk 4]
```diff
@@ -143,15 +141,11 @@ start_trace(Nodes, Patterns, {Procs, Flags}, Options) ->
...
-tracer() -> Result
-
-Equivalent to [`tracer(node())`](`tracer/1`).
-""".
+-doc #{ equiv => tracer(node()) }.
 -spec tracer() -> {ok, [node()]} | {error, term()}.
 tracer() -> tracer(node()).
 -doc """
-tracer(Nodes) -> Result
+tracer(Nodes)
```
Comment (dgud) [inline lib/observer/src/ttb.erl:148]: Remove

File: lib/observer/src/ttb.erl [hunk 5]
```diff
@@ -820,7 +794,7 @@ tpl(A,B,C) ->
     dbg:tpl(A,B,ms(C)).
 
 -doc """
-tpl(Module [, Function [, Arity]], MatchSpec)
+tpl(Module, Function, Arity, MatchSpec)
```
Comment (dgud) [inline lib/observer/src/ttb.erl:797]: Remove?

**Review by dgud — DISMISSED** [review]
I don't know why you have kept some function call examples

**Comment by garazdawi** [comment]
@dgud I kept them because otherwise [the signature will be incorrect](https://www.erlang.org/doc/system/documentation.html#doc-signatures).

File: lib/diameter/src/transport/diameter_sctp.erl [hunk 6]
```diff
@@ -161,7 +161,7 @@ be specified as the value of a transport_module option to
 %% ---------------------------------------------------------------------------
 
 -doc """
-start({Type, Ref}, Svc, [Opt]) -> {ok, Pid, [LAddr]} | {error, Reason}
+start(TypeRef, Svc, Options)
```
[the remark this answers is not in the document]
Comment (garazdawi) [reply lib/diameter/src/transport/diameter_sctp.erl]: fixed.

**Review by bmk — APPROVED** [review]

File: lib/ssh/src/ssh_client_channel.erl [hunk 7]
```diff
@@ -22,16 +22,6 @@
 
 -module(ssh_client_channel).
 -moduledoc """
```
[the remark this answers is not in the document]
Comment (garazdawi) [reply lib/ssh/src/ssh_client_channel.erl:24]: It is not ideal, but IMO better than what was there before? Feel free to continue polishing further :)

[the remark this answers is not in the document]
Comment (u3s) [reply lib/ssh/src/ssh_client_channel.erl:24]: no strong opinion from me. keep it then.

File: lib/ssh/src/ssh_client_channel.erl [hunk 8]
```diff
@@ -248,8 +244,6 @@ call(ChannelPid, Msg, TimeOute) ->
     end.
 
 -doc """
-cast(ChannelRef, Msg) -> ok
-
 Sends an asynchronous message to the channel process and returns ok immediately,
```
Comment (u3s) [inline lib/ssh/src/ssh_client_channel.erl:247]: pls explain. is leaving paragraph containing more than 1 sentence as a signature also fine? In other places I thought single sentence is preferred ...

File: lib/ssh/src/ssh_client_channel.erl [hunk 9]
```diff
@@ -216,8 +214,6 @@ call(ChannelPid, Msg) ->
     call(ChannelPid, Msg, infinity).
 
 -doc """
-call(ChannelRef, Msg, Timeout) -> Reply | {error, Reason}
```
Comment (u3s) [inline lib/ssh/src/ssh_client_channel.erl:219]: while at it. reply/2 and enter_loop/1 docs look strange. can you adjust them as well?

image

below funny italic also is rendered as we have text between `_` characters.
image

**Review by u3s — DISMISSED** [review]

**Pull request merged**